File: Chapter 2-13-Insane Dates/Objective-C/InsaneDates/InsaneDates/ViewController.m
//
//  ViewController.m
//  InsaneDates
//
//  Copyright © 2015 - by Magno Urbano - All rights reserved.
//  Part of the book Troubleshooting Xcode, published by Apress
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  // Do any additional setup after loading the view, typically from a nib.
  
  NSDateComponents *comps = [[NSDateComponents alloc] init];
  [comps setDay:22];
  [comps setMonth:8];
  [comps setYear:2020];
  
  NSCalendar *gregorian = [[NSCalendar alloc] initWithCalendarIdentifier:
                           NSCalendarIdentifierGregorian];
  
  // the magical line
  // enable this line to fix the problem
  // gregorian.timeZone = [NSTimeZone timeZoneForSecondsFromGMT:0];
  
  NSDate *date = [gregorian dateFromComponents:comps];
  
  NSLog(@"Date is %@", date);
  
  
}

- (void)didReceiveMemoryWarning {
  [super didReceiveMemoryWarning];
  // Dispose of any resources that can be recreated.
}

@end

File: Chapter 2-13-Insane Dates/Swift 2.0/Insane Dates/Insane Dates/ViewController.swift
//
//  ViewController.swift
//  Insane Dates
//
//  Copyright © 2015 - by Magno Urbano - All rights reserved.
//  Part of the book Troubleshooting Xcode, published by Apress
//

import UIKit

class ViewController: UIViewController {

  override func viewDidLoad() {
    super.viewDidLoad()
    // Do any additional setup after loading the view, typically from a nib.
    
    let comps   = NSDateComponents()
    comps.day   = 22
    comps.month = 8
    comps.year  = 2020
    
    let gregorian = NSCalendar(identifier: NSCalendarIdentifierGregorian)
    
    // the magical line
    // enable this to fix the problem
    // gregorian?.timeZone = NSTimeZone(forSecondsFromGMT: 0)
    

    let date = gregorian?.dateFromComponents(comps)
    
    
    print("Date is " , date)
    
  }

  override func didReceiveMemoryWarning() {
    super.didReceiveMemoryWarning()
    // Dispose of any resources that can be recreated.
  }


}

